src/Akka.Persistence.Sql/Snapshot/ByteArraySnapshotDao.cs

```diff
@@ -48,6 +107,181 @@ public class ByteArraySnapshotDao : ISnapshotDao
             _readIsolationLevel = snapshotConfig.ReadIsolationLevel;
 
             _shutdownCts = new CancellationTokenSource();
+            _pendingLatestChannel = Channel.CreateUnbounded<LatestSnapRequestEntry>();
+            int maxSubStreamsForReads = 8; // TODO: Configurable
+            int maxRequestsPerBatch = 50;
+            _latestSnapStream = Source.ChannelReader(_pendingLatestChannel.Reader)
+                .GroupBy(maxSubStreamsForReads, a=> a.PersistenceId.GetHashCode()% maxSubStreamsForReads)
+                .BatchWeighted(
+                    maxRequestsPerBatch,
+                    a => 1,
```

Comment:
I know in ShotGlass I had/have a Batcher that could assign a 'zero weight' for cases where a persistence ID was already queued...

FBOW in this first go here I instead decided a flat weight of 1 would be a lot less 'change' but also has -some- benefits for forward progress. (but really mostly because a custom stream stage felt like overkill for first intro of this.)